irods/exception.py

```diff
@@ -61,14 +65,72 @@ class MultipleResultsFound(QueryException):
 
 class iRODSExceptionMeta(type):
     codes = {}
-
+    positive_code_error_message = "For {name}, a positive code of {attrs[code]} was declared."
     def __init__(self, name, bases, attrs):
         if 'code' in attrs:
+            if attrs['code'] > 0:
+                print(self.positive_code_error_message.format(**locals()), file = sys.stderr)
+                exit(1)
```

Comment:
Looking at this again, I'm wondering if this should be an `exit` or another Exception being raised? Maybe I'm misunderstanding where this falls in a "real" use case. I guess this is akin to a compilation/usage error, but then an `exit` feels a little heavy to me.

Reply:
It could be an exception, but probably not that anyone would want to catch -- it indicates a source code error in the irods/exception.py module, one that should be fixed immediately before rerunning the application.  I don't know any parallel cases, either; should we search one out as an example?

Reply:
No, that's okay. I think this is good, just wanted to make sure that was intentional.